### PR TITLE
Add Github Actions for building and testing

### PR DESCRIPTION
**Pull request by ceyhunsen**
# Description

This PR introduces a Github Actions for building and testing Clementine on commits and PR's.

### CODE REVIEW

File: .github/workflows/basic.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    branches: [ "main", "dev" ]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_and_test:
+    name: Compile and test
+
+    runs-on: ubuntu-latest
```
Comment (cetin) [inline .github/workflows/basic.yml:16]: I enabled ubicloud for this. You can use 'ubicloud-standard-4' when running this action.

File: .github/workflows/basic.yml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    - name: Create a wallet in Bitcoin regtest
+      run: bitcoin-27.0/bin/bitcoin-cli -regtest -rpcuser=admin -rpcpassword=admin -rpcport=18443 createwallet "admin"
+    - name: Create funds in Bitcoin regtest
+      run: bitcoin-27.0/bin/bitcoin-cli -regtest -rpcuser=admin -rpcpassword=admin -rpcport=18443 generatetoaddress 101 $(bitcoin-27.0/bin/bitcoin-cli -regtest -rpcuser=admin -rpcpassword=admin -rpcport=18443 getnewaddress)
+
+    - name: Build
+      run: RISC0_DEV_MODE=1 cargo build --verbose
+
+    - name: Create config overwrite file
+      run: |
```
Comment (cetin) [inline .github/workflows/basic.yml:62]: you can update this step like this for better readability

  run: |
    cat << EOF > /home/runner/overwrite.toml
    tracing_debug = "debug,bitcoincore_rpc=info,hyper=error"
    host = "127.0.0.1"
    port = 3000
    secret_key = "5555555555555555555555555555555555555555555555555555555555555555"
    verifiers_public_keys = [
      "4f355bdcb7cc0af728ef3cceb9615d90684bb5b2ca5f859ab0f0b704075871aa",
      "466d7fcae563e5cb09a0d1870bb580344804617879a14949cf22285f1bae3f27",
      "3c72addb4fdf09af94f0c94d7fe92a386a7e70cf8a1d85916386bb2535c7b1b1",
      "2c0b7cf95324a07d05398b240174dc0c2be444d96b159aa6c7f7b1e668680991",
      "9ac20335eb38768d2052be1dbbc3c8f6178407458e51e6b4ad22f1d91758895b"
    ]
     # ... rest of the configuration ...
     
    EOF

**Review by cetin — APPROVED** [review]
I added some comments, but they're not critical. All are for better readability. The rest of them are fine.

**Comment by ceyhunsen** [comment]
> I added some comments, but they're not critical. All are for better readability. The rest of them are fine.

Thanks for that, I applied them. Please do one final check and we will be good to merge. @cetin

**Comment by ceyhunsen** [comment]
Note: It is expected that main branch to fail these tests. Dev branch should pass though.

**Comment by byucesoy** [comment]
Hi folks,

I just sent an e-mail to you, but since your repo is public, I'll write from here as well in case you miss the e-mail.

This is Burak from the Ubicloud team. You might have realized that the workflow job has been waiting in the queue for a long time. It seems we are not able to register our runner to be used in your repository. The process errors out with the following error message.

"Repository level self-hosted runners are disabled on this repository"

Maybe some setting is changed in your organization or repository. If it was intentional, there is no action required, but it also means you won't be able to use Ubicloud runners. I just wanted to let you know about this issue in case the change was not intentional.

**Review by ekrembal — APPROVED** [review]

**Comment by cetin** [comment]
@byucesoy 

Thank you for letting us know. We have realized this issue, and our organization admin is currently making changes to enable self-hosted runners for this repository. We appreciate your prompt notification and support in identifying the problem.

**Review by cetin — APPROVED** [review]

**Pull request merged**